src/script/cpp_api/s_base.cpp

```diff
@@ -180,6 +196,29 @@ void ScriptApiBase::clientOpenLibs(lua_State *L)
 	}
 }
 
+void ScriptApiBase::checkSetByBuiltin()
+{
+	lua_State *L = getStack();
+
+	if (m_gamedef) {
+		lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_READ_VECTOR);
+		luaL_checktype(L, -1, LUA_TFUNCTION);
```

Comment:
the issue with this is that it just throws a random "wrong argument #1 to function ?" error
I suggest
`FATAL_ERROR_IF(lua_type(L, -1) != LUA_TFUNCTION, "missing read_vector");`